### DISCUSSION TITLE
How can I add something into bundle in Rsbuild?  just like Rollup `this.emitFile` hook

### DISCUSSION THREAD

**Discussion by Alkaidcc** [web-infra-dev/rsbuild#2130] [Q&A] [answered]
When I write a rsbuild plugin, I want to have a hook to add something into bundle easily

**Comment by chenjiahan** [comment, marked as answer]
Currently only the `transform` hook of Rsbuild plugin provides an `emitFile` function (see: https://rsbuild.dev/plugins/dev/core#apitransform), we are considering adding a new API for Rsbuild plugin to emit files.

You can also emit files in Rspack plugins, just like the way the webpack plugin emits files.
